Fix error when running with Zarr-Python 3.1.0

The minumum versions workflow is failing with the following (see https://github.com/zarr-developers/VirtualiZarr/actions/runs/21611987811/job/62282604254):

```
Run pixi run -e minimum-versions run-tests-xml-cov
✨ Pixi task (run-tests-xml-cov in minimum-versions): pytest -n auto --run-network-tests --verbose --cov=virtualizarr --cov-report=xml
ImportError while loading conftest '/home/runner/work/VirtualiZarr/VirtualiZarr/conftest.py'.
conftest.py:19: in <module>
    from virtualizarr.manifests import ChunkManifest, ManifestArray
virtualizarr/__init__.py:7: in <module>
    from virtualizarr.xarray import (
virtualizarr/xarray.py:28: in <module>
    from virtualizarr.parsers.typing import Parser
virtualizarr/parsers/__init__.py:7: in <module>
    from virtualizarr.parsers.zarr import ZarrParser
virtualizarr/parsers/zarr.py:12: in <module>
    from zarr.core.dtype import parse_dtype
E   ImportError: cannot import name 'parse_dtype' from 'zarr.core.dtype' (/home/runner/work/VirtualiZarr/VirtualiZarr/.pixi/envs/minimum-versions/lib/python3.12/site-packages/zarr/core/dtype/__init__.py)
```

to fix this, the import from https://github.com/zarr-developers/VirtualiZarr/blob/5e59cde9e60a48f28e2a6dec509cbf3239a9e9da/virtualizarr/parsers/zarr.py#L12 needs to be moved to https://github.com/zarr-developers/VirtualiZarr/blob/5e59cde9e60a48f28e2a6dec509cbf3239a9e9da/virtualizarr/parsers/zarr.py#L185-L186. The tests can be run afterwards to check the fix following https://virtualizarr.readthedocs.io/en/stable/contributing.html with `pixi run -e minimum-versions run-tests-xml-cov`

--- a/virtualizarr/parsers/zarr.py
+++ b/virtualizarr/parsers/zarr.py
@@ -9,7 +9,6 @@ from typing import TYPE_CHECKING, Any, cast
 import zarr
 from obspec_utils.registry import ObjectStoreRegistry
 from zarr.api.asynchronous import open_group as open_group_async
-from zarr.core.dtype import parse_dtype
 from zarr.core.group import GroupMetadata
 from zarr.core.metadata import ArrayV3Metadata
 from zarr.storage import ObjectStore
@@ -184,6 +183,7 @@ class ZarrV2Strategy(ZarrVersionStrategy):
 
         try:
             from zarr.metadata.migrate_v3 import _convert_array_metadata
+            from zarr.core.dtype import parse_dtype
         except (ImportError, AttributeError):
             raise ImportError(
                 f"Zarr-Python>=3.1.3 is required for parsing Zarr V2 into Zarr V3. "

--- a/virtualizarr/parsers/zarr.py
+++ b/virtualizarr/parsers/zarr.py
@@ -182,8 +182,8 @@ class ZarrV2Strategy(ZarrVersionStrategy):
         from zarr.core.metadata import ArrayV2Metadata
 
         try:
-            from zarr.metadata.migrate_v3 import _convert_array_metadata
             from zarr.core.dtype import parse_dtype
+            from zarr.metadata.migrate_v3 import _convert_array_metadata
         except (ImportError, AttributeError):
             raise ImportError(
                 f"Zarr-Python>=3.1.3 is required for parsing Zarr V2 into Zarr V3. "

